pandera/model_components.py

```diff
@@ -0,0 +1,273 @@
+"""SchemaModel components"""
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Iterable,
+    List,
+    Optional,
+    Set,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+    cast,
+)
+
+from .checks import Check
+from .schema_components import Column, Index, PandasDtypeInputTypes, SeriesSchemaBase
+
+AnyCallable = Callable[..., Any]
+SchemaComponent = TypeVar("SchemaComponent", bound=SeriesSchemaBase)
+
+CHECK_KEY = "__check_config__"
+DATAFRAME_CHECK_KEY = "__dataframe_check_config__"
+
+_CheckList = Union[Check, List[Check]]
+
+
+def _to_checklist(checks: Optional[_CheckList]) -> List[Check]:
+    checks = checks or []
+    if isinstance(checks, Check):  # pragma: no cover
+        return [checks]
+    return checks
+
+
+class FieldInfo:
+    """Captures extra information about a field."""
+
+    __slots__ = ("checks", "nullable", "allow_duplicates", "coerce", "regex")
+
+    def __init__(
+        self,
+        checks: Optional[_CheckList] = None,
+        nullable: bool = False,
+        allow_duplicates: bool = True,
+        coerce: bool = False,
+        regex: bool = False,
+    ) -> None:
+        self.checks = _to_checklist(checks)
+        self.nullable = nullable
+        self.allow_duplicates = allow_duplicates
+        self.coerce = coerce
+        self.regex = regex
+
+    def _to_schema_component(
+        self,
+        pandas_dtype: PandasDtypeInputTypes,
+        component: Type[SchemaComponent],
+        checks: _CheckList = None,
+        **kwargs: Any,
+    ) -> SchemaComponent:
+        checks = self.checks + _to_checklist(checks)
+        return component(pandas_dtype, checks=checks, **kwargs)  # type: ignore
+
+    def to_column(
+        self,
+        pandas_dtype: PandasDtypeInputTypes,
+        checks: _CheckList = None,
+        required: bool = True,
+        name: str = None,
+    ) -> Column:
+        """Create a schema_components.Column from a field."""
+        return self._to_schema_component(
+            pandas_dtype,
+            Column,
+            nullable=self.nullable,
+            allow_duplicates=self.allow_duplicates,
+            coerce=self.coerce,
+            regex=self.regex,
+            required=required,
+            name=name,
+            checks=checks,
+        )
+
+    def to_index(
+        self,
+        pandas_dtype: PandasDtypeInputTypes,
+        checks: _CheckList = None,
+        name: str = None,
+    ) -> Index:
+        """Create a schema_components.Index from a field."""
+        return self._to_schema_component(
+            pandas_dtype,
+            Index,
+            nullable=self.nullable,
+            allow_duplicates=self.allow_duplicates,
+            coerce=self.coerce,
+            name=name,
+            checks=checks,
+        )
+
+
+def Field(
+    *,
+    eq: Any = None,
+    neq: Any = None,
```

Comment:
what do you think of renaming this to `ne`? that would be consistent with the [python operator](https://docs.python.org/3/library/operator.html#operator.ne) and [pandas function](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.ne.html).

(fyi I also made alias methods for these in `Check` for convenience #269)

Reply:
Makes sense. I committed the change :+1: